architecture-decisions/0003-preservation.md

```diff
@@ -0,0 +1,67 @@
+# 3. Preservation
+
+Date: 2019-04-02
+
+## Status
+
+Accepted
+
+## Context
+
+We have agreed with our stakeholders that we will preserve our digital objects by packaging them in
+BagIt bags, and storing those bags in cloud storage.  However, there are wide variety of options for
+creating bags, how objects are serialized within the bags, and how they are linked to the digital
+objects.
+
+## Decisions
+
+1. Creating bags
+   1. We will periodically scan for objects which meet the following criteria:
+      1. Status is complete
+      2. Not bagged more recently than their `updated_at` timestamp
+      3. Has a `cloud` preservation policy assigned
+   1. In order to avoid redundant processing when there are multiple updates in a short period of time,
+      we will not trigger bagging from a ChangeSetPersister or other event-driven mechanism.
+1. Packaging bags
+   1. We will export an object, and all of its members recursively, including the metadata for any Valkyrie
+      object it links to (such as collections, controlled vocabulary terms, etc.).
+   1. We will include both files and metadata (serialized as JSON).
+   1. We will package the bag directory in a Tar archive, compressed with GZip.
+   1. A link to the Tar/GZipped bag file will be added to the object's `file_metadata` property.
```

Comment:
Does the object already have a file_metadata or are we creating one? It seems like the process of creating the bag is going to change the updated_at date, triggering another bag creation in the following cycle.

Reply:
I agree that bagging the object will definitely update it.  But I was thinking we would set the `bagged_at` time to the same value as the `updated_at`, so it wouldn't show up as being updated more recently than it was bagged.

Reply:
right! yes I forgot about the `bagged_at` date. `updated_at` is set by valkyrie, i think. I'm not sure how to get them to align precisely.